Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/DocumentSchema.h

```diff
@@ -276,6 +276,18 @@ namespace AZ::DocumentPropertyEditor
                 return {};
             }
 
+            // This case is for handling legacy EnumValueKey attributes
+            if constexpr (AZStd::is_same_v<GenericValueType, uint64_t>)
+            {
+                using EnumConstantBaseType = AZ::SerializeContextEnumInternal::EnumConstantBase;
+                if (auto data = azdynamic_cast<AttributeData<AZStd::unique_ptr<EnumConstantBaseType>>*>(attribute); data != nullptr)
+                {
+                    EnumConstantBaseType* value = static_cast<EnumConstantBaseType*>(data->Get(instance).get());
+                    return ValueToDom(
+                        AZStd::make_pair<GenericValueType, AZStd::string>(value->GetEnumValueAsUInt(), value->GetEnumValueName()));
```

Comment:
Do not pass in argument to `AZStd::make_pair`, it's entire purpose is to deduce the parameters.
I have a change in flight where this will be a compile error and it would require the user to either use `AZStd::pair` with the template arguments specified or AZStd::make_pair without the arguments specified.
```suggestion
                        AZStd::pair<GenericValueType, AZStd::string>(value->GetEnumValueAsUInt(), value->GetEnumValueName()));
```